src/toolkits/object_detection/od_data_iterator.cpp

```diff
@@ -215,24 +219,29 @@ simple_data_iterator::compute_properties(
     std::vector<std::string> expected_class_labels) {
 
   annotation_properties result;
+  gl_sarray classes;
+  if (annotations.size() > 0) {
+    // Construct an SFrame with one row per bounding box.
+    gl_sframe instances;
+    if (annotations.dtype() == flex_type_enum::LIST) {
+      gl_sframe unstacked_instances({{"annotations", annotations}});
+      instances = unstacked_instances.stack("annotations", "bbox",
+                                            /* drop_na */ true);
+    } else {
+      instances["bbox"] = annotations;
+    }
 
-  // Construct an SFrame with one row per bounding box.
-  gl_sframe instances;
-  if (annotations.dtype() == flex_type_enum::LIST) {
-    gl_sframe unstacked_instances({{"annotations", annotations}});
-    instances = unstacked_instances.stack("annotations", "bbox",
-                                          /* drop_na */ true);
-  } else {
-    instances["bbox"] = annotations;
-  }
+    // Extract the label for each bounding box.
+    instances = instances.unpack("bbox", /* column_name_prefix */ "",
+                                 {flex_type_enum::STRING},
+                                 /* na_value */ FLEX_UNDEFINED, {"label"});
 
-  // Extract the label for each bounding box.
-  instances = instances.unpack("bbox", /* column_name_prefix */ "",
-                               { flex_type_enum::STRING },
-                               /* na_value */ FLEX_UNDEFINED, { "label" });
+    // Determine the list of unique class labels,
+    classes = instances["label"].unique().sort();
 
-  // Determine the list of unique class labels,
-  gl_sarray classes = instances["label"].unique().sort();
+    // Record the number of labeled bounding boxes.
+    result.num_instances = instances.size();
+  }
```

Comment:
Could we extract this into a helper function that, given the annotations SArray, returns the list of unique labels and the number of bounding boxes?

Reply:
sounds like a good idea